include/libcyphal/transport/can/transport.hpp

```diff
@@ -0,0 +1,116 @@
+/// @copyright
+/// Copyright (C) OpenCyphal Development Team  <opencyphal.org>
+/// Copyright Amazon.com Inc. or its affiliates.
+/// SPDX-License-Identifier: MIT
+
+#ifndef LIBCYPHAL_TRANSPORT_CAN_TRANSPORT_HPP_INCLUDED
+#define LIBCYPHAL_TRANSPORT_CAN_TRANSPORT_HPP_INCLUDED
+
+#include "media.hpp"
+#include "libcyphal/transport/transport.hpp"
+#include "libcyphal/transport/multiplexer.hpp"
+
+namespace libcyphal
+{
+namespace transport
+{
+namespace can
+{
+
+class ICanTransport : public ITransport
+{};
+
+struct Factory final
+{
+public:
+    Factory() = delete;
+
+    CETL_NODISCARD static inline Expected<UniquePtr<ICanTransport>, FactoryError> make(
```

Comment:
Why can't this be a freestanding function?

Reply:
My rationale was to avoid freestanding functions (which for my taste a bit "c-style"), but instead you know "encapsulate" them into some sort of entity (like this `Factory`). Also, IMHO if we apply such pattern everywhere then it will be easier for users to guess where to find a factory method, like always `<correct_namespace>::Factory::make(...)` something. But in general I'm totally good with essentially the same `<correct_namespace>::makeSomething(...)` approach.

Probably it's again my Swift habits - there are no namespaces in Swift, so any freestanding function is GLOBAL one, which is considered as bad design - so you nest such functions in some (sub)type as static one.

Please let me know if my rationale is not applicable for c++, or there are reasons to specifically go with freestanding alternative. Thanks!

Reply:
will fix this in the next pr